net.solarnetwork.node.setup.web/web/js/graph-data.js

```diff
@@ -0,0 +1,237 @@
+//Author Robert Rewcastle
+//script to draw line graphs of datums with a wattage variable
+//this script will put the graphs inside an html tag with a graphpoint class which 
+//one must put in the html document before loading the script
+//requires d3 https://d3js.org/d3.v3.js
+
+//json field we are interested in
+var key = "watts";
```

Comment:
This script is populating everything into the global JavaScript object. It should be turned into a class-like object instead. See the `datum.js` script for the pattern to follow. Basically create a class for all the functionality that handles the charting, and include a `$(document).ready` hook to use the class if appropriate. For example:

```javascript
'use strict';
SolarNode.DatumCharts = (function() {
			
	var self = {};

	//json field we are interested in
	var key = "watts";
	var units = "W";

	//This map is graphs to look up the latest reading based on their sourceId
	var datamap = {};

	// rest of class here...

	return Object.defineProperties(self, {
		// define public properties & methods here
	});
}());

$(document).ready(function() {
	if ( !SolarNode.isAuthenticated() ) {
		return;
	}
	
	$('#datum-activity-charts').first().each(function() {
		// datum-activity-charts exists so charts should be rendered... 
		SolarNode.DatumCharts.subscribeAndRender();
	});
});
```
```